### PR TITLE
Add auditor pass to ad-hoc codesign all products on Darwin

### PR DESCRIPTION
**Pull request by staticfloat**
Also fixes some tests on v1.6

### CODE REVIEW

File: Manifest.toml [hunk 1]
```diff
@@ -65,7 +62,7 @@ deps = ["Random", "Serialization", "Sockets"]
 uuid = "8ba89e20-285c-5b6f-9357-94700520ee1b"
 
 [[Downloads]]
-deps = ["ArgTools", "LibCURL"]
+deps = ["ArgTools", "LibCURL", "NetworkOptions"]
```
Comment (giordano) [inline Manifest.toml:65]: I think `NetworkOptions` will make the version of Julia we use in CI unhappy

Comment (giordano) [reply Manifest.toml:65]: Also all JLLs as stdlibs

File: test/auditing.jl [hunk 2]
```diff
@@ -27,7 +27,7 @@ end
     @test compatible_marchs(Platform("x86_64", "linux"; march="avx2")) == ["x86_64", "avx", "avx2"]
     @test compatible_marchs(Platform("x86_64", "linux"; march="avx512")) == ["x86_64", "avx", "avx2", "avx512"]
     @test compatible_marchs(Platform("armv7l", "linux")) == ["armv7l"]
-    @test compatible_marchs(Platform("i686", "linux"; march="prescott")) == ["i686", "prescott"]
+    @test compatible_marchs(Platform("i686", "linux"; march="prescott")) == ["pentium4", "prescott"]
```
Comment (giordano) [inline test/auditing.jl:30]: I _think_ the version of Julia we have here requires `i686`, with Julia v1.6 we can do `pentium4`

Comment (staticfloat) [reply test/auditing.jl:30]: Good catches all, this is meant to be merged once we switch over to v1.6.

**Comment by giordano** [comment]
The registry seems out-of-date :grimacing:

File: src/auditor/codesigning.jl [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    if !Sys.isapple(platform)
+        return true
+    end
+
+    rel_path = relpath(path, prefix.path)
+    ur = preferred_runner()(prefix.path; cwd="/workspace/", platform=platform)
+    with_logfile(prefix, "ldid_$(basename(rel_path)).log") do io
+        run(ur, `/usr/local/bin/ldid -S -d $(rel_path)`, io; verbose=verbose)
+    end
+end
```
Comment (giordano) [inline src/auditor/codesigning.jl:22]:
```suggestion
end

```

**Review by giordano — APPROVED** [review]
All green :tada:

File: src/auditor/codesigning.jl [hunk 4]
```diff
@@ -0,0 +1,22 @@
+function check_codesigned(path::AbstractString, platform::AbstractPlatform)
+    # We only perform ad-hoc codesigning on Apple platforms
+    if !Sys.isapple(platform)
+        return true
+    end
+
+    ur = preferred_runner()(dirname(path); cwd="/workspace/", platform=platform)
+    return run(ur, `/usr/local/bin/ldid -d $(basename(path))`)
```
Comment (giordano) [inline src/auditor/codesigning.jl:8]: @staticfloat  Uhm, I think this should have been wrapped in `with_logfile`?

**Pull request merged**